patroni/validator.py

```diff
@@ -406,6 +448,29 @@ def validate(self, name: str) -> Iterator[Result]:
             yield from self._check_executables(path=name)
 
 
+class BinDirectory(Directory):
+    """Check if a Postgres binary directory contains the expected files.
+
+    It is a subclass of :class:`Directory` with an extended capability: translating ``BINARIES`` according to configured
+    ``postgresql.bin_name``, if any.
+
+    :cvar BINARIES: list of executable files that should exist directly under a given Postgres binary directory.
+    """
+
+    BINARIES = ["pg_ctl", "initdb", "pg_controldata", "pg_basebackup", "postgres", "pg_isready", "pg_rewind"]
```

Comment:
The reason we didn't check for `pg_rewind` is that it is nice to have, but it is not required.
Also, it is not available on 9.3 and 9.4 unless someone built/installed it separately.

Reply:
Oh, right! I forgot about this mention in the docs BTW:

> Currently supported PostgreSQL versions: 9.3 to 15.

The first thing that usually comes to my mind is the 5 years cycle of PGDG 😆 

I'll adjust that.